common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -676,16 +676,48 @@ class @Problem
       element = $(element)
       element.find("section[id^='forinput']").removeClass('choicetextgroup_show_correct')
 
-  enableCheckButton: (enable) =>
+  disableAllButtonsWhileRunning: (operationCallback, isFromCheckOperation) =>
+    # Used to keep the buttons disabled while operationCallback is running.
+    @enableAllButtons false, isFromCheckOperation
+    operationCallback().always =>
+      @enableAllButtons true, isFromCheckOperation
+      if isFromCheckOperation
+        @enableCheckButtonAfterResponse
+
+  enableAllButtons: (enable, isFromCheckOperation) =>
+    # Used to enable/disable all buttons in problem.
+    if !isFromCheckOperation
```

Comment:
I think what you want here is to put these @enableCheckButton calls in the if/else between L694-707. So something like:

```coffee
if enable
  # code here to remove the "is-disabled" class and aria attr
  @enableCheckButton true isFromCheckOperation

else
  # code here to *add* the "is-disabled" class and aria attr
  @enableCheckButton false isFromCheckOperation
```